server/handlers/handlers.go

```diff
@@ -55,6 +56,7 @@ type ActivityHandler struct {
 	plugin               PluginIface
 	queue                chan msteams.Activity
 	quit                 chan bool
+	quitting             atomic.Bool
```

Comment:
Introducing a global notion of `quitting` in addition to the per-worker `quit` channel. Note that we don't just `close(quit)` and use that instead of `quitting` since the `ActivityHandler` can be `Start()`ed again as currently designed.

Reply:
I prefer the `close(ch)` idiom to close any listeners. It avoids the mental overhead to keep track of how many consumers are active. One can just close the channel and all listeners would get the signal.

To handle the case of `Start` being called again, we can reinitialize the `quit` channel again.

Reply:
I agree -- in this case, I actually think we can sidestep the requirement to `Start()` more than once in the first place, but it's queued up as a separate investigation: https://mattermost.atlassian.net/browse/MM-55490.

I'll plan to eliminate `quitting` and just do a non-blocking select on the single `quit` channel to detect this at that point.